dmscripts/generate_dos_import_data.py

```diff
@@ -0,0 +1,122 @@
+import os
+import sys
+
+if sys.version_info[0] < 3:
+    import unicodecsv as csv
+else:
+    import csv
+
+
+IMPORT_DATA_CONFIG = {
+    'digital-outcomes-and-specialists': {
+        'lot_names': [
+            'Digital outcomes',
+            'Digital specialists',
+            'User research studios',
+            'User research participants',
+        ],
+        'framework_agreement': {
+            'required_columns': [
+                'supplier_id',
+                'trading_name',
+                'registered_address',
+                'company_number',
+            ]
+        },
+        'result_letter': {
+            'required_columns': [
+                'supplier_declaration_name',
+                'supplier_id',
+            ]
+        }
+    },
+}
+
+
+def reduce_row_to_list(row, selected_indices):
+    result = []
+    for idx, item in enumerate(row):
+        if idx in selected_indices:
+            result.append(item)
+    return result
+
+
+def get_required_column_indices(column_headings, required_column_headings):
+    return [column_headings.index(heading) for heading in required_column_headings]
+
+
+def fill_list_based_on_flags(list_of_flags, potential_values):
+    new_list = []
+    for idx, flag in enumerate(list_of_flags):
+        if bool(int(flag)):
+            new_list.append(potential_values[idx])
+    if len(new_list) < len(potential_values):
+        # pad out array
+        new_list.extend(
+            ['' for entry in range(len(potential_values) - len(new_list))]
+        )
+    return new_list
+
+
+def get_lot_results(values, lot_names):
+    # results are last 3 blocks of 4
+    number_of_lots = len(lot_names)
+    results_by_lot = {
+        'completed': values[:number_of_lots],
+        'failed': values[number_of_lots:(number_of_lots * 2)],
+        'draft': values[(number_of_lots * 2):(number_of_lots * 3)],
+    }
+    for status in results_by_lot:
+        results_by_lot[status] = fill_list_based_on_flags(results_by_lot[status], lot_names)
+    return results_by_lot
+
+
+def get_indices_of_required_columns(supplier_info_headings, required_columns):
+    return [supplier_info_headings.index(heading) for heading in required_columns]
+
+
+def reformat_csv_data(csv_data, framework_slug, document_type):
+    config = IMPORT_DATA_CONFIG[framework_slug]
+    number_of_lots = len(config['lot_names'])
+    lot_results_start_idx = len(csv_data[1]) - (3 * number_of_lots)
+    supplier_info_headings = csv_data[0][:lot_results_start_idx]
+    required_column_indicies = get_indices_of_required_columns(
```

Comment:
The slightly confusing slicing and indexing could be avoided with a [`csv.DictReader`](https://docs.python.org/3/library/csv.html#csv.DictReader).